File: aspnet-core/src/HotelApp.Application.Contracts/Rooms/CreateUpdateRoomDto.cs
using HotelApp.Hotels;
using System;

namespace HotelApp.Rooms
{
    public class CreateUpdateRoomDto
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public int Capacity { get; set; }
        public int MaxNoOfChildren { get; set; }
        public string DefaultPicture { get; set; }

        public decimal Price { get; set; }

        public BedType BedType { get; set; }
        public bool? IsSmoking { get; set; }

        public bool IsBooked { get; set; }

        public Guid HotelId { get; set; }
    }
}

File: angular/src/app/room/models/create-update-room-dto.ts

import { BedType } from './bed-type';
export class CreateUpdateRoomDto  {
  name: string;
  description: string;
  capacity: number;
  maxNoOfChildren: number;
  defaultPicture: string;
  price: number;
  bedType: any;
  isSmoking?: boolean;
  isBooked: boolean;
  hotelId: string;

  constructor(initialValues: Partial<CreateUpdateRoomDto> = {}) {
    if (initialValues) {
      for (const key in initialValues) {
        if (initialValues.hasOwnProperty(key)) {
          this[key] = initialValues[key];
        }
      }
    }
  }
}
